File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/ui/client/pages/OrderHistoryPage.java
package com.mycompany.myshop.testkit.driver.adapter.ui.client.pages;

import com.mycompany.myshop.testkit.driver.adapter.shared.client.playwright.PageClient;

public class OrderHistoryPage extends BasePage {
    private static final String ORDER_NUMBER_INPUT_SELECTOR = "[aria-label='Order Number']";
    private static final String SEARCH_BUTTON_SELECTOR = "[aria-label='Refresh Order List']";
    private static final String ROW_SELECTOR_TEMPLATE = "//tr[contains(., '%s')]";
    private static final String VIEW_DETAILS_LINK_SELECTOR_TEMPLATE = "%s//a[contains(text(), 'View Details')]";

    public OrderHistoryPage(PageClient pageClient) {
        super(pageClient);
    }

    public void inputOrderNumber(String orderNumber) {
        pageClient.fill(ORDER_NUMBER_INPUT_SELECTOR, orderNumber);
    }

    public void clickSearch() {
        pageClient.click(SEARCH_BUTTON_SELECTOR);
    }

    public boolean isOrderListed(String orderNumber) {
        var rowSelector = getRowSelector(orderNumber);
        return pageClient.isVisible(rowSelector);
    }

    public OrderDetailsPage clickViewOrderDetails(String orderNumber) {
        var rowSelector = getRowSelector(orderNumber);
        var viewDetailsLinkSelector = String.format(VIEW_DETAILS_LINK_SELECTOR_TEMPLATE, rowSelector);
        pageClient.click(viewDetailsLinkSelector);
        return new OrderDetailsPage(pageClient);
    }

    private static String getRowSelector(String orderNumber) {
        return String.format(ROW_SELECTOR_TEMPLATE, orderNumber);
    }
}

File: system-test/dotnet/Driver.Adapter/Ui/Client/Pages/OrderHistoryPage.cs
using Driver.Adapter.Shared.Client.Http;

using Driver.Adapter.Shared.Client.Playwright;

using Driver.Port.Dtos;



namespace Driver.Adapter.Ui.Client.Pages;



public class OrderHistoryPage : BasePage

{

    private const string OrderNumberInputSelector = "[aria-label='Order Number']";

    private const string SearchButtonSelector = "[aria-label='Refresh Order List']";



    private const string RowSelectorTemplate = "//tr[contains(., '{0}')]";

    private const string ViewDetailsLinkSelectorTemplate = "{0}//a[contains(text(), 'View Details')]";



    public OrderHistoryPage(PageClient pageClient) : base(pageClient)

    {

    }



    public async Task InputOrderNumberAsync(string? orderNumber)

    {

        await PageClient.FillAsync(OrderNumberInputSelector, orderNumber);

    }



    public async Task ClickSearchAsync()

    {

        await PageClient.ClickAsync(SearchButtonSelector);

    }



    public async Task<bool> WaitForOrderRowAsync(string? orderNumber, int timeoutMilliseconds = 10000)

    {

        try

        {

            var rowSelector = GetRowSelector(orderNumber);

            await PageClient.WaitForVisibleAsync(rowSelector, timeoutMilliseconds);

            return true;

        }

        catch

        {

            return false;

        }

    }



    public async Task<bool> IsOrderListedAsync(string? orderNumber)

    {

        var rowSelector = GetRowSelector(orderNumber);

        return await PageClient.IsVisibleAsync(rowSelector);

    }



    public async Task<OrderDetailsPage> ClickViewOrderDetailsAsync(string? orderNumber)

    {

        var rowSelector = GetRowSelector(orderNumber);

        var viewDetailsLinkSelector = string.Format(ViewDetailsLinkSelectorTemplate, rowSelector);

        await PageClient.ClickAsync(viewDetailsLinkSelector);

        return new OrderDetailsPage(PageClient);

    }



    private static string GetRowSelector(string? orderNumber)

    {

        return string.Format(RowSelectorTemplate, orderNumber);

    }

}









